gramps_webapi/api/resources/emit.py

```diff
@@ -53,7 +53,8 @@ def response(
         args: Optional[Dict] = None,
     ) -> Response:
         """Prepare response."""
-        payload = payload or {}
+        if payload != []:
+            payload = payload or {}
```

Comment:
This is a bit confusing. Can `payload` be anything other than a dict, list, or None? If not, I suggest to type it as `Optional[Union[Dict, List]]` (`Optional[Any]` is equivalent to `Any` anyway). Then
```python
if payload is None:
    payload = {}
```
would be much more explicit.

Reply:
Payload can be an empty dictionary if returning an empty list.  The statement `payload = payload or {}` was causing the empty list to be returned as an empty dictionary which is incorrect.

Reply:
Right, I just found the code
```python
if payload != []:
    payload = payload or {}
```
confusing and thought 
```python
if payload is None:
    payload = {}
```
would be more understandable: the `payload` argument is only modifed when it is `None`, in which case it is set to an empty dict.

Reply:
Oh yeah... yes it is. Thank you, changed.